base/compiler/abstractinterpretation.jl

```diff
@@ -606,6 +606,9 @@ function abstract_call_method(interp::AbstractInterpreter, method::Method, @nosp
     if edge === nothing
         edgecycle = edgelimited = true
     end
+    edge_effects = override_edge_effects(edge_effects, sv, method, :consistent)
```

Comment:
A `:consistent` override on the outer method does not necessarily imply that the statement is `:consistent`. It could be guarded by something.

Reply:
Thanks for pointing this out. We are only able to override `:consistent` edge effect only when the inner `method` is annotated as such.